lib/full-site-editing/block-templates.php

```diff
@@ -122,7 +122,6 @@ function _inject_theme_attribute_in_content( $template_content, $theme ) {
 	foreach ( $template_blocks as $key => $block ) {
 		if (
 			'core/template-part' === $block['blockName'] &&
-			! isset( $block['attrs']['theme'] ) &&
```

Comment:
**Note:**

Removing this line of code lets us inject the current theme stylesheet as a theme attribute, regardless of whether or not a theme attribute already exists on the template part block

Reply:
How is this different to what #28131 does? It appears this forces the block attribute to always have the same value. Isn't that the same as removing support altogether?

Reply:
> How is this different to what #28131 does? 

From what I understand about #28131, it attempts to do away with the theme attribute entirely. (Definitely let me know if I'm misreading!)

With these code changes, the theme attribute itself is still being utilized. Instead, we:
  - strip theme attributes provided by file-based templates 
  - replace it with the current theme's stylesheet
  
We're still relying on theme attributes to query for and to load templates/ template parts.

Reply:
> the theme attribute itself is still being utilized.

> We're still relying on theme attributes

Is it? Are you? If I am running a theme `my-theme`, then any of these will end up resolving in the same way:

```
<!-- wp:template-part {"slug":"header","theme":"my-theme"} /-->
```
```
<!-- wp:template-part {"slug":"header"} /-->
```
```
<!-- wp:template-part {"slug":"header","theme":"I can put anything at all in here and it will resolve the same"} /-->
```

So, the question is, what effect does the attribute have?

Reply:
I'm trying to make sure I understand you correctly, so this answer is a little verbose. Just did some quick tests in my local environment to double check, but here's what I see when working with theme provided, file-based templates.

### Without this PR applied:
```
// Loads as long as the theme we're referencing lives in the root of the theme directory (wp-content/themes)
<!-- wp:template-part {"slug":"header","theme":"my-theme"} /-->

// Loads properly because of the changes in #28088
<!-- wp:template-part {"slug":"header"} /--> // 

// Does not load
<!-- wp:template-part {"slug":"header","theme":"I can put anything at all in here and it will resolve the same"} /-->
```

### With this PR applied:
```
// Loads properly
<!-- wp:template-part {"slug":"header","theme":"my-theme"} /-->

// Loads properly
<!-- wp:template-part {"slug":"header"} /--> // 

// Loads properly
<!-- wp:template-part {"slug":"header","theme":"I can put anything at all in here and it will resolve the same"} /-->
```

I believe you're trying to comment on what happens when we apply this PR, and you're saying that, regardless of what we use as a theme attribute, the file based template part still loads. It appears as if the theme attribute no longer matters. 

If this is what you mean, then yes, the intention of these code changes is to ignore any theme attributes defined in file-based templates. These changes, however, only affects **file-based** templates (see [here](https://github.com/WordPress/gutenberg/blob/2cf8c1c081408400ed6a395e5f1722ec34541665/lib/full-site-editing/block-templates.php#L152-L159)), and wouldn't change how **customized** template theme attributes are handled (see [here](https://github.com/WordPress/gutenberg/blob/2cf8c1c081408400ed6a395e5f1722ec34541665/lib/full-site-editing/block-templates.php#L186)). That is to say, we wouldn't be stripping theme attributes from user-customized and saved templates.

In that sense, we're not removing support for block attributes.

Reply:
@jeyip Your explanation makes total sense and you did understand correctly, although I'm not sure "stripping" is the correct description of what this patch does. It's more a case of "overwriting" or "overriding" the value of the attribute with the active theme.

You clarified this wouldn't apply to user-customised templates, but I've noticed that the present version of the master branch actually has no UI to vary the value from the active theme anyway. It can only be done through code view.

Furthermore, as I mentioned in [this comment](https://github.com/WordPress/gutenberg/pull/28131#issuecomment-759824272), it would be very difficult to build a useful UI for varying the theme value:

> The problem with reinstating the theme reference into the endpoint is it would make it extremely difficult to list out the "active" template-parts (in the site editor, for example) without listing out every template-part from every theme.

In other words, by continuing support for the theme attribute, you're only providing immediate support for users who know how to use the editor's code view, and no one presently knows how any potential future UI will work or whether it will eventuate at all.

I guess it's up to you whether you're OK with this, but it seems to me like a dysfunctional way to run the project. Maybe you weren't aware of some of these points.

Reply:
> although I'm not sure "stripping" is the correct description of what this patch does. It's more a case of "overwriting" or "overriding" the value of the attribute with the active theme.

Good point! I changed the PR title to reflect this.

> You clarified this wouldn't apply to user-customised templates, but I've noticed that the present version of the master branch actually has no UI to vary the value from the active theme anyway. It can only be done through code view.

This makes sense to me. Currently, the theme attribute is set through the "code view" in the editor, and is always assigned the currently active theme.

> Furthermore, as I mentioned in this comment, it would be very difficult to build a useful UI for varying the theme value:

I'm not sure I follow. 🤔  Could you elaborate on why we'd want to vary the theme value? I think you're talking about allowing the user themselves to assign theme attributes to the template part, but I'm not certain.

Reply:
> Could you elaborate on why we'd want to vary the theme value?

Ostensibly, for requesting template parts from themes other than the active one.